src/python/pants/backend/js/target_types.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    Dependencies,
+    MultipleSourcesField,
+    SingleSourceField,
+    Target,
+    TargetFilesGenerator,
+)
+
+JS_FILE_EXTENSIONS = (".js",)
+
+
+class JSSourceField(SingleSourceField):
+    expected_file_extensions = JS_FILE_EXTENSIONS
+
+
+class JSGeneratorSourcesField(MultipleSourcesField):
+    expected_file_extensions = JS_FILE_EXTENSIONS
+
+
+# -----------------------------------------------------------------------------------------------
+# `js_source` and `js_sources` targets
+# -----------------------------------------------------------------------------------------------
+
+
+class JSSourceTarget(Target):
+    alias = "js_source"
```

Comment:
`js_source` or `javascript_source`? My vote is `js` - I think it's unambiguous and less annoying to type.

--

Kinda makes me wish we did `py_source`. Oh well

Reply:
Yeah, probably..And the idea is `tailor` automates most of it for you